src/instances.ts

```diff
@@ -343,11 +343,11 @@ const addAssetHooks = !!webpack.version!.match(/^4.*/)
       // compilation is actually of type webpack.Compilation, but afterProcessAssets
       // only exists in webpack5 and at the time of writing ts-loader is built using webpack4
       const makeAssetsCallback = (compilation: any) => {
-        compilation.hooks.afterProcessAssets.tap('ts-loader', () =>
+        compilation.hooks.processAssets.tap({ name: 'ts-loader', stage: webpack.Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL }, (_assets: webpack.Asset[]) => {
```

Comment:
I took a gamle on the webpack.Asset. Guess I need to double check with the debugger.

Reply:
Not even close 😅
![image](https://user-images.githubusercontent.com/5993909/107020835-a32e9000-67a3-11eb-9894-311df0ddd94a.png)

Reply:
@johnnyreilly: Since Webpack doesn't provide types for everything, are we just supposed to add what we know and improve/change later if we find more information?
Right now I know that it's a `Record<string, CachedSource>`

But could potentially be other sources as well:
![image](https://user-images.githubusercontent.com/5993909/107021361-539c9400-67a4-11eb-856d-11bc9176f004.png)

Reply:
> Since Webpack doesn't provide types for everything, 

It doesn't? What's missing? It's built with TypeScript (albeit derived from JSDoc) so I'd imagined it all should be there